hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/ContainerBalancerTask.java

```diff
@@ -341,6 +341,16 @@ public List<ContainerBalancerTaskIterationStatusInfo> getCurrentIterationsStatis
         .max()
         .orElse(0);
 
+    // the balancing thread could go to sleep without having initialized findTargetStrategy, so we check for null here
+    Map<DatanodeDetails, Long> sizeEnteringNodes = Collections.emptyMap();
+    if (findTargetStrategy != null) {
+      sizeEnteringNodes = findTargetStrategy.getSizeEnteringNodes();
+    }
+    Map<DatanodeDetails, Long> sizeLeavingNodes = Collections.emptyMap();
```

Comment:
Maybe it will be better to create empty map only if findSourceStrategy == null? 
```
    if (findSourceStrategy == null) {
      sizeLeavingNodes = Collections.emptyMap();
    } else {
      sizeLeavingNodes = findSourceStrategy.getSizeLeavingNodes();
    }
```

Reply:
Sure - we can come back to this after we've resolved your first review comment above.